File: composeApp/src/androidMain/kotlin/com/github/eylulnc/notia/feature/history/ui/HistoryUiState.kt
package com.github.eylulnc.notia.feature.history.ui

data class HistoryMonthGroup(
    val monthLabel: String, // "October 2023"
    val items: List<HistoryItem>
)

data class HistoryUiState(
    val months: List<HistoryMonthGroup>,
    val currentStreak: Int,
    val longestStreak: Int,
    val isLoading: Boolean = false
) {
    val isEmpty: Boolean get() = months.isEmpty()

    companion object {
        fun loading() = HistoryUiState(
            months = emptyList(),
            currentStreak = 0,
            longestStreak = 0,
            isLoading = true
        )
    }
}

File: iosApp/iosApp/Features/History/HistoryUiState.swift
import Foundation

struct HistoryItem: Identifiable {
    let id = UUID()
    let dateLabel: String  // "Mon, Jan 22"
    let dayAbbrev: String  // "Mon"
    let dayNumber: String  // "22"
    let focusText: String
}

struct HistoryMonthGroup: Identifiable {
    let id = UUID()
    let monthLabel: String  // "January 2026"
    let items: [HistoryItem]
}

struct HistoryUiState {
    var months: [HistoryMonthGroup]
    var currentStreak: Int
    var longestStreak: Int
    var isLoading: Bool

    var isEmpty: Bool { months.isEmpty }

    static func loading() -> HistoryUiState {
        HistoryUiState(months: [], currentStreak: 0, longestStreak: 0, isLoading: true)
    }
}
